auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiAdapterFactoryProcessor.java

```diff
@@ -108,69 +109,88 @@
         })
         .collect(toList());
 
-    if (!elements.isEmpty()) {
-      for (Element element : adapterFactories) {
-        if (!element.getModifiers().contains(ABSTRACT)) {
-          error(element, "Must be abstract!");
-        }
-        // Safe to cast because this is only applicable on types anyway
-        TypeElement type = (TypeElement) element;
-        if (!implementsJsonAdapterFactory(type)) {
-          error(element, "Must implement JsonAdapter.Factory!");
-        }
-        String adapterName = classNameOf(type);
-        PackageElement packageElement = packageElementOf(type);
-        String packageName = packageElement.getQualifiedName().toString();
-
-        List<TypeElement> applicableElements = elements.stream()
-            .filter(e -> {
-              Visibility typeVisibility = Visibility.ofElement(e);
-              switch (typeVisibility) {
-                case PRIVATE:
+    if (elements.isEmpty()) {
+      Element reportableElement = adapterFactories.iterator().next();
+      if (!autoValueElements.isEmpty()) {
+        processingEnv.getMessager().printMessage(ERROR,
+            "Failed to write JsonAdapter.Factory: Cannot generate class for this "
+                + "@MoshiAdapterFactory-annotated element because while @AutoValue-annotated "
+                + "elements were found on the compilation classpath, none of them contain a "
+                + "requisite public static JsonAdapter-returning signature method to opt in to "
```

Comment:
Does this need to be updated since this is now the legacy method for including AVM types?

Reply:
Not exactly, this does say those are what's used to opt-in to this

Reply:
I don't see anything here about the `@JsonClass` annotation.  That would be the preferred method of making the class available to `@MoshiAdapterFactory` over the static JsonAdapter-returning method, right?

Reply:
Nvm, understood that if using `@JsonClass` you don't need to use this factory, since Moshi already knows that.